.github/workflows/build.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and test
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  create:
+
+jobs:
+  build-and-test-code:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
```

Comment:
do you mind to add a comment on this line @JBAhire?

Reply:
yup. classic 2 hour waste so why not 😅

Reply:
Digging a level deeper, why do we need all branches and tags? The only case I can think of is the publish job in order to generate the version. If that's the only reason, we could leave the default fetch depth elsewhere and keep the speedup right?

Reply:
Default fetch depth was giving this error: 
`Unexpected error while parsing HEAD commit: Missing commit 393a1fddea86f8a239acfe877dce7375630bb3b9`

https://github.com/hypertrace/attribute-service/runs/1509280728?check_suite_focus=true#step:8:30

so I changed it to 0 but now I have set it to 2 which is working fine.

Reply:
interesting - that sounds like it might actually be a force push issue (since I can't find that commit either, it doesn't belong to any branches).

The potential concern with any non-0 number in my mind was version generation, which walks the git history to find the most recent tag, and uses that to generate the next version. So, while a number like 2 may work in that case now, in future builds, it may result in incorrect versioning. That incorrect versioning could also be subtle, as it may revert it back to the initial version number without failing.

On the other hand, a job that doesn't produce artifacts shouldn't really care about the version - so a depth of 1 (the default) I'd expect to work there.

Given that and the likelihood for people to copy/paste without a full understanding, maybe leaving the depth of 0 and updating the comment to give the "why" (e.g. to bring down the commit history + tags for use in version calculation) would be sufficient.